### ISSUE TITLE
Failing to edit dataset's metadata

### ISSUE THREAD

**Issue by heshamkelbashandy** [heshamkelbashandy/ess-dive-catalog#58]
In GitLab by @elbashandy on Feb 2, 2018, 12:50

Failed to edit fields 'Project' and 'DOE Contracts' in a dataset on my localhost as shown here:

![Screen_Shot_2018-02-02_at_9.47.17_AM](/uploads/a81ed7c85fb8b5d7cc431cae220419bc/Screen_Shot_2018-02-02_at_9.47.17_AM.png)

**Steps to reproduce the issue:**
*  Launch the ess-dive-catalog application on your localhost. Follow the steps here: https://gitlab.com/ess-dive/ess-dive-catalog/blob/master/README.md
*  Sign in with your Orcid.
*  Upload the attached dataset file, file the needed overview information and then click 'Submit Data'.
*  Try to edit fields 'Project' and 'DOE Contracts' for the uploaded dataset (attached) and click 'Submit'.

This is the output of 'get_example.py':

```
Traceback (most recent call last):
  File "./scripts/get_example.py", line 37, in <module>
    r = essdive.mn.read.meta_get(identifier)
  File "/Users/absho/workspace/lbnl/ess_dive/ess-dive-toolset/essdive/mn/read.py", line 61, in meta_get
    return _get(identifier, ENDPOINT_META)
  File "/Users/absho/workspace/lbnl/ess_dive/ess-dive-toolset/essdive/mn/read.py", line 156, in _get
    raise MNApiException(response)
essdive.mn.MNApiException: HTTP ERROR 404: <?xml version="1.0" encoding="UTF-8"?><error detailCode="1800" errorCode="404" name="NotFound">    <description>No system metadata could be found for given PID: d21e754e-b4e9-471e-8ef5-e975664e3d0f</description></error>
```

Output for `get_example.py urn:uuid:d21e754e-b4e9-471e-8ef5-e975664e3d0f`:
```
<?xml version="1.0" ?>
<ns1:systemMetadata xmlns:ns1="http://ns.dataone.org/service/types/v2.0">
  <serialVersion>0</serialVersion>
  <identifier>urn:uuid:d21e754e-b4e9-471e-8ef5-e975664e3d0f</identifier>
  <formatId>eml://ecoinformatics.org/eml-2.1.1</formatId>
  <size>1951</size>
  <checksum algorithm="MD5">bf089545a94b1a5a566d7380d7f70c6a</checksum>
  <submitter>http://orcid.org/0000-0001-5044-9453</submitter>
  <rightsHolder>http://orcid.org/0000-0001-5044-9453</rightsHolder>
  <accessPolicy>
    <allow>
      <subject>public</subject>
      <permission>read</permission>
    </allow>
  </accessPolicy>
  <replicationPolicy replicationAllowed="false"/>
  <archived>false</archived>
  <dateUploaded>2018-02-01T23:54:06.513Z</dateUploaded>
  <dateSysMetadataModified>2018-02-03T01:08:56.089Z</dateSysMetadataModified>
  <originMemberNode>urn:node:mnTestESS_DIVE</originMemberNode>
  <authoritativeMemberNode>urn:node:mnTestESS_DIVE</authoritativeMemberNode>
  <fileName>Third Dataset.xml</fileName>
</ns1:systemMetadata>


Search by ID: urn:uuid:d21e754e-b4e9-471e-8ef5-e975664e3d0f
id=urn:uuid:d21e754e-b4e9-471e-8ef5-e975664e3d0f
	resourceMap=[u'urn:uuid:66cadde7-c087-43b8-bf1a-770248d3e709']
	documents=[u'urn:uuid:f6440ea0-2b0e-4a4d-84da-479d4a34b75e']
	obsoletes=
	obsoletedBy=
	isDocumentedBy=
```

[resourceMap_urn_uuid_6ec405bd-c092-4115-978a-9b1b04cae4e0.xml.zip](/uploads/2ce73a030f6926498c90004137bed043/resourceMap_urn_uuid_6ec405bd-c092-4115-978a-9b1b04cae4e0.xml.zip)

**Comment by heshamkelbashandy** [comment]
In GitLab by @elbashandy on Feb 2, 2018, 12:50

added ~3203215 label

**Comment by heshamkelbashandy** [comment]
In GitLab by @vchendrix on Feb 2, 2018, 12:57

@elbashandy The pid is missing a prefix `urn:uuid:d21e754e-b4e9-471e-8ef5-e975664e3d0f`.

Try `get_example.py urn:uuid:d21e754e-b4e9-471e-8ef5-e975664e3d0f` to see if your are still getting an error.  If not then try to update the permissions on the object.

**Comment by heshamkelbashandy** [comment]
In GitLab by @vchendrix on Feb 2, 2018, 13:01

However the DOE contract number problem is an issue. I will review that.

**Comment by heshamkelbashandy** [comment]
In GitLab by @elbashandy on Feb 2, 2018, 13:19

changed the description

**Comment by heshamkelbashandy** [comment]
In GitLab by @elbashandy on Feb 2, 2018, 13:20

changed the description

**Comment by heshamkelbashandy** [comment]
In GitLab by @vchendrix on Feb 2, 2018, 15:22

@elbashandy I am not able to reproduce this error.  What browser, browser version and OS version are you using?

**Comment by heshamkelbashandy** [comment]
In GitLab by @elbashandy on Feb 4, 2018, 21:50

I am using Google Chrome Version 63.0.3239.132 (Official Build) (64-bit), MacOS High Sierra Version 10.13.2

**Comment by heshamkelbashandy** [comment]
In GitLab by @vchendrix on Feb 5, 2018, 12:44

@elbashandy Try `get_example.py urn:uuid:d21e754e-b4e9-471e-8ef5-e975664e3d0f` with an up-to-date authentication token.

**Comment by heshamkelbashandy** [comment]
In GitLab by @elbashandy on Feb 5, 2018, 12:54

changed the description

**Comment by heshamkelbashandy** [comment]
In GitLab by @vchendrix on Feb 5, 2018, 13:45

@elbashandy Is this a problem for all datasets that you create or just this one?  The reason I am asking is that I cannot reproduce this issue with the given information.  Are any javascript errors thrown?

**Comment by heshamkelbashandy** [comment]
In GitLab by @elbashandy on Feb 5, 2018, 13:52

Yes with all of them. Just tested it now with another dataset and I opened the JS console and it's throwing this error. This happened when I tried to edit the "Project" Field.

![Screen_Shot_2018-02-05_at_1.51.38_PM](/uploads/f9549b32006371a9da079f465499efc9/Screen_Shot_2018-02-05_at_1.51.38_PM.png)

**Comment by heshamkelbashandy** [comment]
In GitLab by @vchendrix on Feb 23, 2018, 15:59

closed via commit 2d7c23f8129882b834d936f13f50072d3d0c8a22

**Comment by heshamkelbashandy** [comment]
In GitLab by @vchendrix on Feb 23, 2018, 15:59

closed via commit 517700eb539b28b92abe21d6f4747ba854ccd15a
